custom_components/mcp23017/binary_sensor.py

```diff
@@ -70,24 +80,35 @@ def int_config(pins):
         return gpinten
 
     def update_sensors(port):
-        for pin_num in mcp.int_flag:
-            _LOGGER.debug("Dispatching MCP23017: {} on port: {} ".format(
-                           DEVICE.format(i2c_address, pin_num),port))
-            dispatcher_send(hass, DEVICE.format(i2c_address, pin_num))
-        mcp.clear_ints()
+        # TODO: port is probably a bad name, most likely this can be the interrupt pin?
+        _LOGGER.debug("MCP23017 interrupt detected.")
+        for mcpDict in mcpDicts:
+            mcp = mcpDict.get("mcp")
+            i2c_address = mcpDict.get("i2c_address")
+            #intflag = mcp.int_flag
+            #for pin_num in intflag: # this gets all pins that triggered the interrupt, and then only updates the pins that actually changed interrupt, but this works unreliably
+            for pin_num in range(16): # update all pins instead
+                _LOGGER.debug("Dispatching MCP23017 change notification: {} on RPI pin: {} ".format(
+                               DEVICE.format(i2c_address, pin_num),port))
+                dispatcher_send(hass, DEVICE.format(i2c_address, pin_num))
+            mcp.clear_ints()
```

Comment:
The intflag way is preferred, it would be awesome if we could get this to work reliably

Reply:
We would need to find out what causes the reliability issues, do you have any ideas on how to troubleshoot it?
Also, here we don't need to update all the 16 pins but only the ones defined in the configuration file like:
```
for pin_num in mcpDict.get("pins"):
```
But it would definitively be better if we can make it work reliably using the int_flag.

I can probably find some time to work on it this coming weekend.